### PR TITLE
Enhancement duplicate checking

### PR DESCRIPTION
**Pull request by YongbinWang**
Implemented duplicate checking, to be further tested when catching for bugs. This addresses #247 , #239 , #222.
image

List Command update to display accurate output for empty list. This addresses #246 .
image

Added some JUnit test for decrement atomic habits.

### CODE REVIEW

File: src/main/java/wellnus/atomichabit/command/AddCommand.java [hunk 1]
```diff
@@ -78,7 +95,7 @@ protected String getFeatureKeyword() {
      * After that, print a message telling the user what the new habit added is
```
Comment (haoyangw) [inline src/main/java/wellnus/atomichabit/command/AddCommand.java:95]: Update JavaDoc to include `@throws` description on the new `AtomicHabitException`

File: src/test/java/wellnus/atomichabit/AtomicHabitTest.java [hunk 2]
```diff
@@ -89,6 +89,21 @@ public void addHabit_invalidCommand_atomicHabitExceptionThrown() {
...
+     *
+     * @throws WellNusException
+     */
+    @Test
+    public void addHabit_duplicateHabit_atomicHabitExceptionThrown() throws WellNusException {
+        String payload = "junit test";
+        String testAddCommand = String.format("%s --name %s", ADD_HABIT_COMMAND, payload + System.lineSeparator());
+        HashMap<String, String> arguments = parser.parseUserInput(testAddCommand);
+        Command addCommand = new AddCommand(arguments, habitList);
+        addCommand.execute();
```
Comment (haoyangw) [inline src/test/java/wellnus/atomichabit/AtomicHabitTest.java:103]: Hmm i think this is unnecessary cos the next line(`asertThrows`) already calls this method for you. Then you can avoid adding the `throws WellnusException` clause for this method

**Review by haoyangw — APPROVED** [review]
Hmm some minor nitpicks to make the code better. Otherwise, tested and works fine, so LGTM!

File: src/test/java/wellnus/atomichabit/AtomicHabitTest.java (hunk 2 above)
Comment (YongbinWang) [reply src/test/java/wellnus/atomichabit/AtomicHabitTest.java:103]: Require this part to simulate adding 2 duplicate habits.

Comment (haoyangw) [reply src/test/java/wellnus/atomichabit/AtomicHabitTest.java:103]: My bad, misunderstood the intent here. resolved!

**Comment by haoyangw** [comment]
All good now, merging!

File: src/main/java/wellnus/atomichabit/command/AddCommand.java (hunk 1 above)
Comment (YongbinWang) [reply src/main/java/wellnus/atomichabit/command/AddCommand.java:95]: Resolved in 38c723b.

**Pull request merged**